app/src/main/java/com/duckduckgo/app/browser/downloader/NetworkFileDownloader.kt

```diff
@@ -129,6 +132,26 @@ class NetworkFileDownloader @Inject constructor(
             setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED)
         }
         val manager = context.getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager?
+
+        // In some unknown cases enqueuing a download on Android Q throws an IllegalArgumentException. We want the user to still
+        // be able to use the app and inform them the download has failed. If this will happen in the future on other versions we want
+        // to see the crashes.
+        if (appBuildConfig.sdkInt == Build.VERSION_CODES.Q) {
+            startDownloadOnAndroidQ(manager, request, callback)
```

Comment:
I think we should probably not make a distinction between `Q` and any other versions.
I have taken a look at AOSP and this can potentially still happen in all versions. The `ContentResolver.insert` method throws `IllegalArgumentException` if "the provider doesn't support any of the bundled arguments" which come from the `request`.
The exception is never `try-catch`ed inside AOSP but also the signature is never carried on.

Reply:
Thanks, updated to do this for all Android versions. I re-checked and we only got this crash on Android 10 and there are some people mentioning this being noticed on Android 10 only, however nobody managed to understand how / why is happening. 
There are a few different download-related crashes that indeed happen on all OS versions and they'll be investigated separately.